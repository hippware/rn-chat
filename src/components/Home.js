import React from 'react';
import {View, Image, AppState, StyleSheet, NetInfo, InteractionManager, Animated, ScrollView, TouchableOpacity, Text, Dimensions} from 'react-native';
import {Actions} from 'react-native-router-native';
import FilterBar from './FilterBar';
import FilterTitle from './FilterTitle';
import {k, backgroundColorDay, backgroundColorNight} from '../globals';
export const HEIGHT = Dimensions.get('window').height;
export const WIDTH = Dimensions.get('window').width;
import NavBarCloseButton from './NavBarCloseButton';
import assert from 'assert';
import BotButton from './BotButton';
import Chats from './ChatListView';
import Map from './Map';
import location from '../store/locationStore';
import model from '../model/model';
import * as xmpp from '../store/xmpp/xmpp';
import EventList from './EventListView';
import {observer} from 'mobx-react/native';
import {autorun} from 'mobx';
import Notification from './Notification';
import autobind from 'autobind-decorator';
import profileStore from '../store/profileStore';

import {MessageBar, MessageBarManager} from 'react-native-message-bar';
import PushNotification from 'react-native-push-notification';
import TransparentGradient from './TransparentGradient';
import ParallaxScrollView from 'react-native-parallax-scroll-view';

class OwnMessageBar extends MessageBar {
    componentWillReceiveProps(nextProps) {}
}

@autobind
@observer
export default class Home extends React.Component {
    constructor(props) {
        super(props);
        this.contentOffsetY = 0;
        this.state = {
            top: new Animated.Value(0),
        };
    }

    componentDidMount() {
        AppState.addEventListener('change', this._handleAppStateChange);
        NetInfo.addEventListener('change', this._handleConnectionInfoChange);
        MessageBarManager.registerMessageBar(this.refs.alert);
    }

    componentWillUnmount() {
        AppState.removeEventListener('change', this._handleAppStateChange);
        NetInfo.removeEventListener('change', this._handleConnectionInfoChange);
        MessageBarManager.unregisterMessageBar();
    }

    tryReconnect() {
        if (model.registered && model.connected === false && !model.connecting && model.user && model.password && model.server) {
            console.log('TRYING RECONNECT');
            profileStore.connect();
        }
    }

    _handleConnectionInfoChange(connectionInfo) {
        console.log('CONNECTIVITY:', connectionInfo);
        if (connectionInfo !== 'none') {
            this.tryReconnect();
        }
    }

    _handleAppStateChange(currentAppState) {
        console.log('CURRENT APPSTATE:', currentAppState);
        // reconnect automatically
        if (currentAppState === 'active') {
            this.tryReconnect();
        }
        if (currentAppState === 'inactive') {
            globalStore.finish();
            xmpp.disconnect();
        }
    }

    onScroll(event) {
        // switch nav bar is scroll position is below threshold
        // this.contentOffsetY = event.nativeEvent.contentOffset.y;
        // if (event.nativeEvent.contentOffset.y > 90 * k) {
        //   if (!this.state.hideActivityBar) {
        //     this.setState({hideActivityBar: true});
        //   }
        // } else {
        //   if (this.state.hideActivityBar) {
        //     this.setState({hideActivityBar: false});
        //   }
        // }
    }

    componentWillMount() {
        PushNotification.setApplicationIconBadgeNumber(0);
        // this.handler = autorun(()=> {
        //   console.log("REFRESH BADGE", model.chats.unread, model.friends.newFollowers.length);
        //   for (let key of ['home_', 'friendsMain', 'myAccount_']){
        //     Actions.refresh({key,
        //       rightButtons: [{badgeValue: `${model.chats.unread}`}],
        //       leftButtons: [{badgeValue: `${model.friends.newFollowers.length}`}]
        //     });
        //   }
        // });
    }

    componentWillUnmount() {
        if (this.handler) {
            this.handler();
            this.handler = null;
        }
    }

    scrollTo(num) {
        InteractionManager.runAfterInteractions(() => {
            Animated.timing(
                // Uses easing functions
                this.state.top, // The value to drive
                {toValue: num} // Configuration
            ).start();
        });
    }

    render() {
        if (this.props.fullMap && !this.state.fullMap) {
            // animate
            InteractionManager.runAfterInteractions(() => {
                this.setState({fullMap: true});

                Animated.timing(
                    // Uses easing functions
                    this.state.top, // The value to drive
                    {toValue: HEIGHT} // Configuration
                ).start();
            });
        }
        if (!this.props.fullMap && this.state.fullMap) {
            // animate
            InteractionManager.runAfterInteractions(() => {
                this.setState({fullMap: false});
                if (location.location) {
                    this._map.setCenterCoordinate(location.location.latitude, location.location.longitude);
                    this._map.setZoomLevel(17);
                }
                Animated.timing(
                    // Uses easing functions
                    this.state.top, // The value to drive
                    {toValue: 0} // Configuration
                ).start();
            });
        }
        const backgroundColor = location.isDay ? backgroundColorDay : backgroundColorNight;
        // console.log("RENDER HOME, isDay:", location.isDay, location.location);
        return (
            <View style={{flex: 1}}>
                <Map
                    ref={map => {
                        this._map = map;
                    }}
                    followUser
                    fullMap={this.props.fullMap}
                    location={location.location}
                    isDay={location.isDay}
                />
                <Animated.View style={{flex: 1, transform: [{translateY: this.state.top}]}}>
                    <EventList ref='list' />
                </Animated.View>
                {!this.state.fullMap && <BotButton />}
                <OwnMessageBar ref='alert' />
            </View>
        );
        //   <FilterBar style={{paddingLeft:15*k, paddingRight:15*k}}
        // hidden={this.state.hideActivityBar}
        // isDay={location.isDay}>
        // <Text key="all">All</Text>
        // <Text key="friends">Friends</Text>
        //
        //   </FilterBar>
    }
}
