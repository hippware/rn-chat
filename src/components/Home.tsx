import React from 'react'
import {View} from 'react-native'
import {observer} from 'mobx-react/native'
import autobind from 'autobind-decorator'
import BotButton from './BotButton'
import EventList from './EventListView'
import Connectivity from './Connectivity'
import {Actions} from 'react-native-router-flux'

@autobind
@observer
export default class Home extends React.Component<{}> {
  private eventList: any

  private scrollToTop() {
    this.eventList && this.eventList.wrappedInstance.scrollToTop()
  }

  componentDidMount() {
    setTimeout(() => {
      // Actions.friendsMain();
      // Actions.botDetails({item: '80927e0c-18d8-11e8-a5ab-0a580a020223'});
      // Actions.botShareSelectFriends({botId: '80927e0c-18d8-11e8-a5ab-0a580a020223'});
      // Actions.followed();
    }, 500)
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <EventList ref={ref => (this.eventList = ref)} />
        <BotButton />
        <Connectivity />
      </View>
    )
  }
}
