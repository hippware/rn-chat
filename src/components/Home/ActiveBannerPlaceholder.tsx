import React from 'react'
import {View, Image} from 'react-native'

import {k} from '../Global'
import {RText} from '../common'
import {colors} from '../../constants'

const placeholderImg = require('../../../images/geoBotPlaceholder.png')

const ActiveBannerPlaceholder = () => {
  return (
    <View
      style={{
        backgroundColor: 'white',
        paddingLeft: 25 * k,
        paddingBottom: 15 * k,
        flex: 0,
        alignSelf: 'flex-start'
      }}
    >
      <Image source={placeholderImg} style={{width: 72 * k, height: 79 * k}} resizeMode="contain" />
      <RText
        size={13}
        style={{textAlign: 'center', left: -5 * k}}
        numberOfLines={2}
        color={colors.DARK_GREY}
      >
        {'No Current\r\nVisitors'}
      </RText>
    </View>
  )
}

export default ActiveBannerPlaceholder
