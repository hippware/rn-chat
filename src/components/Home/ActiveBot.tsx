import React, {Component} from 'react'
import {TouchableOpacity, View, FlatList, StyleSheet, Text, Image} from 'react-native'

import {observer, inject} from 'mobx-react/native'
import {observable} from 'mobx'
import Swipeable from 'react-native-swipeable'
import LinearGradient from 'react-native-linear-gradient'

import BotBubble from '../map/BotBubble'
import {k} from '../Global'
// import Avatar from '../ProfileAvatar'
import {Avatar} from '../common'

@inject('wocky')
@observer
class ActiveBot extends React.Component<{bot: any}> {
  @observable profile: any
  @observable visitors: Array<any>

  arr = [1, 2, 3]
  obj = {thing: 1, other: 2}

  async componentWillMount() {
    console.log('visitors', this.props.bot.visitors)
    this.profile = await this.props.wocky.getProfile(this.props.bot.owner.id)
  }

  render() {
    console.log('active', this.profile && this.profile.toJSON())
    return (
      <View style={{padding: 15}}>
        <BotBubble bot={this.props.bot} scale={0} />
        {/* <VisitorHeads /> */}
        {this.profile && (
          <View style={{position: 'absolute', top: 0, right: 0}}>
            <Avatar profile={this.profile} tappable={false} size={20} />
          </View>
        )}
      </View>
    )
  }
}

export default ActiveBot
