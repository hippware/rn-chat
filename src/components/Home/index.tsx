import React from 'react'
import {View} from 'react-native'
import {observer} from 'mobx-react/native'
import BotButton from '../BotButton'
import EventList from './EventListView'
import Connectivity from '../Connectivity'
import {Actions} from 'react-native-router-flux'

@observer
class Home extends React.Component<{}> {
  eventList: any

  scrollToTop() {
    if (this.eventList) this.eventList.wrappedInstance.scrollToTop()
  }

  componentDidMount() {
    setTimeout(() => {
      // Actions.friendsMain()
      // Actions.botDetails({item: '80927e0c-18d8-11e8-a5ab-0a580a020223'})
      // Actions.botShareSelectFriends({botId: '80927e0c-18d8-11e8-a5ab-0a580a020223'})
      // Actions.followed()
    }, 500)
  }

  render() {
    return (
      <View style={{flex: 1}} testID="screenHome">
        <EventList ref={ref => (this.eventList = ref)} />
        <BotButton />
        <Connectivity />
      </View>
    )
  }
}

export default Home
