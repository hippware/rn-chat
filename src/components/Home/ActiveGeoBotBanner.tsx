import React from 'react'
import {
  View,
  FlatList,
  StyleSheet,
  Animated,
  TouchableOpacity,
  Image,
  ViewStyle,
} from 'react-native'
import {observer, inject} from 'mobx-react/native'
import {colors} from '../../constants'
import ActiveGeofenceBot from './ActiveGeofenceBot'
import HeaderLocationOverlay from './HeaderLocationOverlay'
import ActiveBannerPlaceholder from './ActiveBannerPlaceholder'
import {IBot, IWocky} from 'wocky-client'
import {analyticsGeoWidgetTap} from '../../utils/analytics'
import {k, isIphoneX, isIphone, minHeight} from '../Global'
import {Actions} from 'react-native-router-flux'
import InvisibleModeOverlay from './InvisibleModeOverlay'
import {settings} from '../../globals'
import {INavStore} from '../../store/NavStore'
import {IHomeStore} from '../../store/HomeStore'
import {computed} from 'mobx'
import {ILocationShare} from 'third-party/wocky-client/src/model/LocationShare'
import {getType} from 'mobx-state-tree'
import ActiveLocationSharer from './ActiveLocationSharer'

type Props = {
  wocky?: IWocky
  navStore?: INavStore
  homeStore?: IHomeStore
  analytics?: any
  enabled: boolean
}

type State = {
  yOffset: Animated.Value
}

export interface IActiveBannerItem {
  outerStyle: ViewStyle
  innerStyle: ViewStyle
}

@inject('wocky', 'analytics', 'homeStore', 'navStore')
@observer
export default class ActiveGeoBotBanner extends React.Component<Props, State> {
  state: State = {
    yOffset: new Animated.Value(0),
  }

  @computed
  get bannerItems(): Array<ILocationShare | IBot> {
    const {profile, activeBots} = this.props.wocky!
    return profile ? [...profile.locationSharers.list.slice(), ...activeBots.slice()] : []
  }

  componentWillReceiveProps(newProps) {
    if (newProps.enabled !== this.props.enabled) {
      const hide = !newProps.enabled
      Animated.spring(this.state.yOffset, {
        toValue: hide ? -180 : 0,
        // speed: 6,
      }).start()
    }
  }

  render() {
    const {wocky, navStore, homeStore} = this.props
    return (
      <Animated.View
        style={{
          transform: [{translateY: this.state.yOffset}],
        }}
        onStartShouldSetResponder={() => {
          this.props.analytics.track(analyticsGeoWidgetTap)
          return false
        }}
        pointerEvents="box-none"
      >
        <View
          style={{
            backgroundColor: 'white',
            paddingTop: isIphoneX ? 28 * k : isIphone ? 23 * k : 0,
            shadowColor: homeStore!.mapType === 'hybrid' ? '#333' : colors.GREY,
            shadowOffset: {width: 0, height: 2},
            shadowOpacity: 1,
            shadowRadius: 5,
          }}
        >
          <FlatList
            data={this.bannerItems}
            horizontal
            keyExtractor={this.keyExtractor}
            renderItem={this.renderActiveBot}
            showsHorizontalScrollIndicator={false}
            ListEmptyComponent={<ActiveBannerPlaceholder />}
            style={{paddingLeft: 8 * k}}
          />
          {!wocky!.connected && <View style={styles.overlay} />}
          <HeaderLocationOverlay />
          <InvisibleModeOverlay />
        </View>
        {navStore!.scene !== 'botCompose' && !homeStore!.fullScreenMode && (
          <Buttons mapType={homeStore!.mapType} />
        )}
      </Animated.View>
    )
  }

  keyExtractor = item => item.id

  renderActiveBot = ({item}: {item: IBot | ILocationShare}) =>
    getType(item).name === 'Bot' ? (
      <ActiveGeofenceBot bot={item as IBot} outerStyle={styles.outer} innerStyle={styles.inner} />
    ) : (
      <ActiveLocationSharer
        sharer={item as ILocationShare}
        outerStyle={styles.outer}
        innerStyle={styles.inner}
      />
    )
}

const Buttons = ({mapType}) => (
  <View
    style={{
      marginRight: 10,
      marginTop: 15,
      alignItems: 'center',
      alignSelf: 'flex-end',
    }}
    pointerEvents="box-none"
  >
    <TouchableOpacity
      onPress={() => Actions.bottomMenu()}
      onLongPress={() => settings.allowDebugScreen && Actions.debugScreen()}
    >
      <Image source={settingsImg} />
    </TouchableOpacity>
    <TouchableOpacity style={{marginTop: 15}} onPress={() => Actions.attribution()}>
      <Image source={mapType === 'hybrid' ? infoImgWhite : infoImg} />
    </TouchableOpacity>
  </View>
)

const settingsImg = require('../../../images/settingsBtn.png')
const infoImgWhite = require('../../../images/iButtonWhite.png')
const infoImg = require('../../../images/info.png')
const dotWidth = 12

const styles = StyleSheet.create({
  overlay: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: colors.addAlpha(colors.WHITE, 0.7),
  },
  outer: {
    marginHorizontal: 7.5 * k,
    paddingVertical: 15,
    width: 71,
    alignItems: 'center',
    marginTop: 6 * minHeight,
  },
  inner: {
    width: 75,
    alignItems: 'center',
    justifyContent: 'center',
  },
  newDot: {
    position: 'absolute',
    top: -5,
    right: -5,
    borderColor: 'white',
    borderWidth: 2,
    borderRadius: dotWidth / 2,
    width: 10,
    height: 10,
    backgroundColor: colors.GOLD,
  },
})
