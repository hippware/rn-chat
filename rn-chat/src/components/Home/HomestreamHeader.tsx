import React from 'react'
import {View} from 'react-native'

import ActiveGeoBotBanner from './ActiveGeoBotBanner'

const HomeStreamHeader = props => (
  <View>
    <ActiveGeoBotBanner {...props} />
  </View>
)

export default HomeStreamHeader
