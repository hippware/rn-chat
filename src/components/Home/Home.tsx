import React from 'react'
import {View} from 'react-native'
import {observer} from 'mobx-react-lite'
import MapHome from './MapHome'
import HorizontalCardList from './HorizontalCardList'
import ActiveGeoBotBanner from './ActiveGeoBotBanner'
import {IHomeStore} from '../../store/HomeStore'
import {INavStore} from '../../store/NavStore'
import {width, height} from '../Global'
import {inject} from 'mobx-react'
import alert from 'src/utils/alert'

type Props = {
  homeStore?: IHomeStore
  navStore?: INavStore
  name: string
  userId?: string
}

const Home = inject('homeStore', 'navStore')(
  observer(({homeStore, navStore, name, userId}: Props) => {
    const {fullScreenMode, setIndex, list, index} = homeStore!
    const isCurrent = navStore!.scene === name
    return (
      <View
        style={{
          width,
          height,
          justifyContent: 'space-between',
        }}
        testID="screenHome"
      >
        <MapHome />
        <ActiveGeoBotBanner enabled={!fullScreenMode && isCurrent} />
        {isCurrent && (
          <HorizontalCardList
            setIndex={setIndex}
            list={list.slice()}
            index={index}
            enabled={!fullScreenMode && isCurrent}
          />
        )}
      </View>
    )
  })
)

export default Home
