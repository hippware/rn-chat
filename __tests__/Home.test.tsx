import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Home from '../src/components/Home/Home'
import {Provider} from 'mobx-react/native'
// import injects from './utils/inject-mocks'
// import wocky from './utils/mockWocky'

describe('Home', () => {
  it('renders with no data', () => {
    const wocky = {
      createChat: () => {
        /*noop*/
      },
      profile: {
        handle: 'jerkham',
        firstName: 'eric',
        lastName: 'kirkham',
        email: 'eric.kirkham@gmail.com',
        loaded: true,
        updateError: '',
      },
    }
    // NOTE: individual events don't get rendered in these snapshots...see *.test.js for each event type
    const toRender = (
      <Provider wocky={wocky}>
        <Home name="home" />
      </Provider>
    )
    const tree = renderer.create(toRender).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
