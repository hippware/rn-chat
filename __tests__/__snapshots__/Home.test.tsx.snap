// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders with no data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
  testID="screenHome"
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "bottom": -100,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../images/loader_pink.gif",
        }
      }
      style={
        Array [
          undefined,
          Object {
            "height": 72,
            "width": 72,
          },
        ]
      }
    />
  </View>
  <View
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "paddingTop": 56,
          "shadowColor": "rgb(212,212,212)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 5,
        }
      }
    >
      <RCTScrollView
        ListEmptyComponent={<ActiveBannerPlaceholder />}
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "paddingLeft": 16,
          }
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-start",
                      "flex": 0,
                      "paddingHorizontal": 16,
                      "paddingVertical": 12,
                    },
                    Object {
                      "marginLeft": 6,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../images/NewPin.png",
                    }
                  }
                  style={
                    Object {
                      "height": 62,
                      "marginLeft": -6.6000000000000005,
                      "width": 56,
                    }
                  }
                />
                <Text
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "color": "rgb(253,95,108)",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 13,
                      },
                      Object {
                        "left": -10,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  New!
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "flex": 0,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/PlaceholderPin.png",
                  }
                }
                style={
                  Object {
                    "height": 57,
                    "marginTop": 5,
                    "width": 51,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "flex": 0,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/PlaceholderPin.png",
                  }
                }
                style={
                  Object {
                    "height": 57,
                    "marginTop": 5,
                    "width": 51,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "flex": 0,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/PlaceholderPin.png",
                  }
                }
                style={
                  Object {
                    "height": 57,
                    "marginTop": 5,
                    "width": 51,
                  }
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
      <BlurView
        blurAmount={10}
        blurType="xlight"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "bottom": 0,
              "flexDirection": "row",
              "left": 0,
              "padding": 40,
              "paddingHorizontal": 60,
              "paddingTop": 70,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../images/footOpaquePink.png",
            }
          }
          style={
            Object {
              "height": 89,
              "marginRight": 40,
              "width": 69,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            SHARE YOUR VISITS!
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(155,155,155)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 13,
                },
                Object {
                  "marginTop": 10,
                },
              ]
            }
          >
            Know when your friends visit your favorite places!
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "borderColor": "rgb(253,95,108)",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "marginTop": 20,
                "opacity": 1,
                "padding": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgb(253,95,108)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
            >
              GIVE LOCATION ACCESS
            </Text>
          </View>
        </View>
      </BlurView>
      <BlurView
        blurAmount={10}
        blurType="xlight"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "bottom": 0,
              "flexDirection": "row",
              "left": 0,
              "padding": 40,
              "paddingHorizontal": 60,
              "paddingTop": 70,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../images/footOpaquePink.png",
            }
          }
          style={
            Object {
              "marginRight": 40,
              "width": 42,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            See visits to your favorite locations!
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "borderColor": "rgb(253,95,108)",
                "borderRadius": 2,
                "borderWidth": 0.5,
                "marginTop": 20,
                "opacity": 1,
                "padding": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgb(253,95,108)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
            >
              Turn Off Invisible Mode
            </Text>
          </View>
        </View>
      </BlurView>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "marginRight": 10,
          "marginTop": 15,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/settingsBtn.png",
            }
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 15,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/info.png",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
